src/cpp/CMakeLists.txt

```diff
@@ -76,6 +76,7 @@ if(UNIX)
 
    # gcc hardending options (see: http://wiki.debian.org/Hardening)
    if(NOT APPLE)
+      add_definitions(-O2)
```

Comment:
Without this, I was getting warnings related to the use of `_FORTIFY_SOURCE` without an optimization level set.